examples/flux_operator_ca_hpa/application_ca_hpa_metrics.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+
+from kubernetes import client as k8s
+from kubernetes import watch, config
+from datetime import datetime
+import json
+import os 
+import argparse
+import threading
+
+import kubescaler.utils as utils
+
+# Save data here
+here = os.path.dirname(os.path.abspath(__file__))
+
+# Create data output directory
+data = os.path.join(here, "data")
+
+def datetime_utcnow_str():
+    return datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+def str_to_datetime(datetime_str_obj):
+    return datetime.strptime(datetime_str_obj, "%Y-%m-%dT%H:%M:%SZ")
+
+def watch_for_pod_events(k8s_client, output_dir, namespace='flux-operator'):
+    
+    pods_for_flux = {}
+    watcher = watch.Watch()
+    for event in watcher.stream(func=k8s_client.list_namespaced_pod, namespace=namespace):
+        event_type = event['type']
+        object = event['object']  # object is one of type return_type
+        raw_object = event['raw_object']  # raw_object is a dict
+
+        name = raw_object["metadata"]["name"]
+        status = raw_object['status']
+
+        if event_type == "ADDED":
+            pods_for_flux[name] = {}
+            pods_for_flux[name]['created'] = raw_object['metadata']['creationTimestamp']
+            if raw_object['status']['phase'] == "Running":
+                pods_for_flux[name]["waiting_for_scheduled"] = {"current_status": True, "start": None, "end": None, "duration": None}
+                pods_for_flux[name]["waiting_for_container"] = {"current_status": True, "start": None, "end": None, "duration": None}
+            else:
+                pods_for_flux[name]["waiting_for_scheduled"] = {"current_status": False, "start": datetime_utcnow_str(), "end": None, "duration": None}
+                pods_for_flux[name]["waiting_for_container"] = {"current_status": False, "start": datetime_utcnow_str(), "end": None, "duration": None}
+        
+        elif event_type == "MODIFIED":
+            for condition in status['conditions']:
+                if condition['type'] == 'PodScheduled' and condition['status'] == 'True' and (not pods_for_flux[name]["waiting_for_scheduled"]["current_status"]):
+                    pods_for_flux[name]["waiting_for_scheduled"]["current_status"] = True
+                    pods_for_flux[name]['waiting_for_scheduled']['end'] = datetime_utcnow_str()
+                    pods_for_flux[name]['waiting_for_scheduled']['duration'] = (datetime.utcnow() - str_to_datetime(pods_for_flux[name]['waiting_for_scheduled']['start'])).total_seconds()
+                elif condition['type'] == 'Initialized':
```

Comment:
You could also do:

```python
elif condition['type'] in ['Initialized', 'Ready', 'ContainersReady']:
    pass
```

I would put an else that at least prints a warning in case a condition type is added/comes up that we do not expect!